exchanges/okcoin/okcoin_websocket.go

```diff
@@ -147,584 +247,690 @@ func (o *OKCoin) WsHandleData(respRaw []byte) error {
+	err := json.Unmarshal(respRaw, &resp)
 	if err != nil {
 		return err
 	}
-
-	a := o.GetAssetTypeFromTableName(response.Table)
-	trades := make([]trade.Data, len(response.Data))
-	for i := range response.Data {
-		f := strings.Split(response.Data[i].InstrumentID, delimiterDash)
-		c := currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
-
-		tSide, err := order.StringToOrderSide(response.Data[i].Side)
-		if err != nil {
-			o.Websocket.DataHandler <- order.ClassificationError{
-				Exchange: o.Name,
-				Err:      err,
-			}
-		}
-
-		amount := response.Data[i].Size
-		if response.Data[i].Quantity != 0 {
-			amount = response.Data[i].Quantity
-		}
-		trades[i] = trade.Data{
-			Amount:       amount,
-			AssetType:    a,
-			CurrencyPair: c,
-			Exchange:     o.Name,
-			Price:        response.Data[i].Price,
-			Side:         tSide,
-			Timestamp:    response.Data[i].Timestamp,
-			TID:          response.Data[i].TradeID,
-		}
-	}
-	return trade.AddTradesToBuffer(o.Name, trades...)
-}
-
-// wsProcessCandles converts candle data and sends it to the data handler
-func (o *OKCoin) wsProcessCandles(respRaw []byte) error {
-	var response WebsocketCandleResponse
-	err := json.Unmarshal(respRaw, &response)
+	cp, err := currency.NewPairFromString(resp.Arg.InstrumentID)
 	if err != nil {
 		return err
 	}
 
-	a := o.GetAssetTypeFromTableName(response.Table)
-	for i := range response.Data {
-		f := strings.Split(response.Data[i].InstrumentID, delimiterDash)
-		c := currency.NewPairWithDelimiter(f[0], f[1], delimiterDash)
-
-		timeData, err := time.Parse(time.RFC3339Nano,
-			response.Data[i].Candle[0])
-		if err != nil {
-			return fmt.Errorf("%v Time data could not be parsed: %v",
-				o.Name,
-				response.Data[i].Candle[0])
-		}
-
-		candleIndex := strings.LastIndex(response.Table, okcoinWsCandle)
-		candleInterval := response.Table[candleIndex+len(okcoinWsCandle):]
-
-		klineData := stream.KlineData{
-			AssetType: a,
-			Pair:      c,
-			Exchange:  o.Name,
-			Timestamp: timeData,
-			Interval:  candleInterval,
-		}
-		klineData.OpenPrice, err = strconv.ParseFloat(response.Data[i].Candle[1], 64)
-		if err != nil {
-			return err
-		}
-		klineData.HighPrice, err = strconv.ParseFloat(response.Data[i].Candle[2], 64)
-		if err != nil {
-			return err
-		}
-		klineData.LowPrice, err = strconv.ParseFloat(response.Data[i].Candle[3], 64)
+	algoOrder := make([]order.Detail, len(resp.Data))
+	for x := range resp.Data {
+		var side order.Side
+		side, err := order.StringToOrderSide(resp.Data[x].Side)
 		if err != nil {
 			return err
 		}
-		klineData.ClosePrice, err = strconv.ParseFloat(response.Data[i].Candle[4], 64)
+		var oType order.Type
+		oType, err = order.StringToOrderType(resp.Data[x].OrderType)
 		if err != nil {
 			return err
 		}
-		klineData.Volume, err = strconv.ParseFloat(response.Data[i].Candle[5], 64)
+		var status order.Status
+		status, err = order.StringToOrderStatus(resp.Data[x].State)
 		if err != nil {
 			return err
 		}
```

Comment:
Instead of using `return err` here, could you use something like:
```go
	b.Websocket.DataHandler <- order.ClassificationError{
					Exchange: b.Name,
					OrderID:  notification.Data[i].OrderID,
					Err:      err,
				}
```
This is mostly because rather than returning an error for order 1 of 10 (for example) its ultimately best to know about orders, even if they don't have the correct order type (for example) . At least then we can query the order via orderID and get the right info